amgut/lib/data_access/ag_data_access.py

```diff
@@ -969,3 +969,303 @@ def is_deposited_ebi(self, barcode):
+
+    def get_random_supplied_kit_id_scanned_unconsented(self):
+        """ Returns randomly chosen supplied_kit_id and barcode which has been
+        scanned but is without consent.
+        For unit testing only!
+
+        Returns
+        -------
+        str: [supplied_kit_id, barcode]
+            example: ['fNIYa', '000001053']
+
+        Raises
+        ------
+        ValueError
+            If no kits can be found in the DB that have been scanned and
+            are without consent."""
+        with TRN:
+            sql = """SELECT supplied_kit_id, barcode
+                     FROM barcodes.barcode
+                     JOIN ag.ag_kit_barcodes USING (barcode)
+                     JOIN ag.ag_kit USING (ag_kit_id)
+                     WHERE barcodes.barcode.scan_date IS NOT NULL
+                     AND ag.ag_kit_barcodes.survey_id IS NULL
+                     LIMIT 1"""
+            TRN.add(sql, [])
+            info = TRN.execute_fetchindex()
+            if not info:
+                raise ValueError('No kits found.')
+            return info[0]
+
+    def get_random_handout_printed_min6_supplied_kit_id(self):
+        """ Returns a random supplied_kit_id with printed results
+        and 6 swaps per kit.
+        For unit testing only!
+
+        Returns
+        -------
+        supplied_kit_id : str
+            A supplied_kit_id. Example: 'DS_ubdvq'
+
+        Raises
+        ------
+        ValueError
+            If no hand out kit exists, satisfing the given conditions."""
+        with TRN:
+            sql = """SELECT kit_id
+                     FROM ag.ag_handout_kits
+                     WHERE swabs_per_kit = 6 AND print_results = TRUE"""
+            TRN.add(sql, [])
+            info = TRN.execute_fetchindex()
+            if not info:
+                raise ValueError('No kits found.')
+            return info[0][0]
+
+    def get_random_email(self):
+        """ Return randomly chosen email.
+        For unit testing only!
+
+        Returns
+        -------
+        str: email
+            Example: 'a03E9u6ZAu@glA+)./Vn'
+
+        Raises
+        ------
+        ValueError
+            If no emails be found in the DB."""
+        with TRN:
+            sql = """SELECT email
+                     FROM ag.ag_login
+                     LIMIT 1"""
+            TRN.add(sql, [])
+            info = TRN.execute_fetchindex()
+            if not info:
+                raise ValueError('No emails found.')
+            return info[0][0]
+
+    def get_random_barcode(self, deposited=True):
+        """ Returns randomly chosen barcode.
+        For unit testing only!
+
+        Parameters
+        ----------
+        deposited : boolean
+            If true, pick a deposited barcode. Default = True
+
+        Returns
+        -------
+        str: barcode
+            Example: '000032951'
+
+        Raises
+        ------
+        ValueError
+            If no barcodes can be found in the DB."""
+        depos = 'TRUE'
```

Comment:
cool. I didn't know this and wanted to be on the save side. Removed those three lines.